src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentComplexAttributeContentPass.cs

```diff
@@ -78,44 +76,26 @@ private static void ProcessAttribute(IntermediateNode parent, IntermediateNode n
             }
             else
             {
-                removeNode = true;
                 issueDiagnostic = true;
             }
         }
         else if (node.Children is [CSharpCodeIntermediateNode])
         {
             // This is the case when an attribute contains a code block @{ ... }
             // We don't support this.
-            removeNode = true;
-            issueDiagnostic = true;
-        }
-        else if (node.Children is [CSharpExpressionIntermediateNode, HtmlContentIntermediateNode { Children: [IntermediateToken { Content: "." }] }])
-        {
-            // This is the case when an attribute contains something like "@MyEnum."
-            // We simplify this to remove the "." so that tooling can provide completion on "MyEnum"
-            // in case the user is in the middle of typing
-            node.Children.RemoveAt(1);
```

Comment:
@davidwengier do we have tooling tests covering this scenario? Don't want to accidentally regress anything.

Reply:
Added an integration test, not at all sure I wrote it correctly, though.